driver-core/src/main/com/mongodb/internal/session/ServerSessionPool.java

```diff
@@ -77,57 +72,46 @@ public ServerSessionPool(final Cluster cluster, @Nullable final ServerApi server
 
     public ServerSession get() {
         isTrue("server session pool is open", !closed);
-        ServerSessionImpl serverSession = serverSessionPool.get();
-        while (shouldPrune(serverSession)) {
-            serverSessionPool.release(serverSession, true);
-            serverSession = serverSessionPool.get();
+        ServerSessionImpl serverSession = available.pollLast();
```

Comment:
As discussed, keep pruning until there is a non-stale session or the pool is empty